include/kernel.h

```diff
@@ -135,35 +135,11 @@ struct k_mem_partition;
 enum k_objects {
 	K_OBJ_ANY,
 
-	/* Core kernel objects */
-	K_OBJ_ALERT,
-	K_OBJ_MSGQ,
-	K_OBJ_MUTEX,
-	K_OBJ_PIPE,
-	K_OBJ_SEM,
-	K_OBJ_STACK,
-	K_OBJ_THREAD,
-	K_OBJ_TIMER,
-	K_OBJ__THREAD_STACK_ELEMENT,
-
-	/* Driver subsystems */
-	K_OBJ_DRIVER_ADC,
-	K_OBJ_DRIVER_AIO_CMP,
-	K_OBJ_DRIVER_COUNTER,
-	K_OBJ_DRIVER_CRYPTO,
-	K_OBJ_DRIVER_DMA,
-	K_OBJ_DRIVER_FLASH,
-	K_OBJ_DRIVER_GPIO,
-	K_OBJ_DRIVER_I2C,
-	K_OBJ_DRIVER_I2S,
-	K_OBJ_DRIVER_IPM,
-	K_OBJ_DRIVER_PINMUX,
-	K_OBJ_DRIVER_PWM,
-	K_OBJ_DRIVER_ENTROPY,
-	K_OBJ_DRIVER_RTC,
-	K_OBJ_DRIVER_SENSOR,
-	K_OBJ_DRIVER_SPI,
-	K_OBJ_DRIVER_UART,
+	/* Enumeration values are generated during build by
+	 * gen_kobject_list.py.  It includes basic kernel objects (e.g.  pipes
+	 * and mutexes) and driver types.
+	 */
+#include <kobj-types-enum.h>
```

Comment:
Doxygen is looking for this include file when generating API documentation.  Either we need an empty file as a place holder in the folder structure (maybe with just a comment saying how it will be generated), or we need to tell doxygen to just ignore this include with:

    /** @cond 
     *  Doxygen should ignore this build-time generated include file when
     *  generating API documentation
     */

    #include <kobj-types-enum.h>

    /** @endcond
     */